### ISSUE TITLE
Migrate ObservableArray to createObservableArray

### ISSUE THREAD

**Issue by samreid** [phetsims/axon#336]
From https://github.com/phetsims/axon/issues/330, it is time to convert all usages of ObservableArray to use the Proxy-based implementation createObservableArray.  This will affect many repos.

**Comment by samreid** [comment]
I completed the migration on my working copy.  I ran the following checks:

* Aqua fuzzing is passing
* PhET-iO Aqua fuzzing is passing
* Comparing the phet-io-api before/after for Energy Forms and Changes looks as expected (new elementAddedEmitter, elementRemovedEmitter, and removed ObservableArray events and methods).
* I tested `energyFormsAndChanges.introScreen.model.air.energyChunkWanderControllers.lengthProperty` in studio and saw that it incremented as expected.
* Lint-everything is passing.
* Studio=>Launch seemed OK for Energy Forms and Changes
* Natural Selection state wrapper seems operational

I'll commit shortly.
